cpmpy/solvers/ortools.py

```diff
@@ -97,17 +86,17 @@ def __add__(self, cons):
         """
         # store new user vars
         new_user_vars = get_variables(cons)
-        for v in frozenset(new_user_vars)-frozenset(self.user_vars):
-            self.user_vars.append(v)
+        for v in frozenset(new_user_vars) - frozenset(self.user_vars):
+            self.user_vars.add(v)
 
-        flat_cons = flatten_constraint(cons)
+        flat_cons = only_bv_implies(flatten_constraint(cons))
         # add new (auxiliary) variables
```

Comment:
solver_var now creates new variables on the fly so this little code-block no longer needed

Reply:
True, also fixed for minizinc.